paddle/gserver/gradientmachines/RecurrentGradientMachine.cpp

```diff
@@ -1399,8 +1399,8 @@ void RecurrentGradientMachine::createDataOutlinkCopySizeInfo(
              getBeamSize() > 1 ? finalPaths_.size() : finalPaths_[0].size());
     int* starts = inputSeqStartPos->getMutableData(false);
     int seqId = 0;
-    for (int i = 0; i < finalPaths_.size(); ++i) {
-      for (int j = 0; j < finalPaths_[i].size(); ++j) {
+    for (size_t i = 0; i < finalPaths_.size(); ++i) {
+      for (size_t j = 0; j < finalPaths_[i].size(); ++j) {
```

Comment:
Don't use `size_t` in `for` loop: https://google.github.io/styleguide/cppguide.html#Integer_Types

You can write like this:
```cpp
for (int i = 0; i < int(finalPaths_.size()); ++i) {
  for (int j = 0; j < int(finalPaths_[i].size()); ++j) {
    // ...
  }
}
```